src/main/java/org/jboss/set/ValidationCommand.java

```diff
@@ -0,0 +1,147 @@
+        Map<String, Collection<String>> projectComponentsMap = new HashMap<>();
+        Set<String> participants = new HashSet<>();
+        boolean valid = true;
+
+        for (LotteryConfig.Participant participant : lotteryConfig.participants()) {
+
+            if (participant.maxIssues() != null && participant.maxIssues() <= 0) {
+                errorAndPrint("Participant %s has defined non-positive number of max issues".formatted(participant.email()));
+                valid = false;
+            }
+
+            for (LotteryConfig.Participant.Project project : participant.projects()) {
+                if (participants.contains(participant.email())) {
+                    errorAndPrint("Two identical participants detected %s".formatted(participant.email()));
+                }
+                participants.add(participant.email());
+
+                if (project.participation().maxIssues() <= 0) {
+                    errorAndPrint("Participant %s has defined non-positive number of max issues for project %s"
+                            .formatted(participant.email(), project.project()));
+                    valid = false;
+                }
+
+                // we assume a valid project has some components
+                Collection<String> components = projectComponentsMap.computeIfAbsent(project.project(), projectKey -> {
+                    Project jiraProject = getJiraEndpoint().getClient().getProjectClient().getProject(projectKey).claim();
+                    return StreamSupport.stream(jiraProject.getComponents().spliterator(), false).map(BasicComponent::getName)
+                            .map(String::toLowerCase).toList();
+                });
+
+                if (components.isEmpty()) {
+                    errorAndPrint(
+                            "For participant %s and project %s no components were found on JIRA. A typo in project is thus expected."
+                                    .formatted(participant.email(), project.project()));
+                    valid = false;
+                }
+
+                if (project.components() == null) {
+                    continue;
+                }
+
+                Collection<String> lowerCasedComponents = project.components().stream().map(String::toLowerCase).toList();
+                if (!components.containsAll(lowerCasedComponents)) {
+                    errorAndPrint(
+                            "For participant %s and project %s some components were found under the project based on JIRA. A typo in one or more components is thus expected."
+                                    .formatted(participant.email(), project.project()));
+                    valid = false;
+                }
+            }
+        }
+        return valid;
+    }
+
+    private void errorAndPrint(String message) {
+        spec.commandLine().getOut().println(message);
+        Log.error(message);
+    }
+
+    private void errorAndPrint(Throwable e, String message) {
```

Comment:
switch arguments because this is how it's usually done in similar frameworks and this order will be unexpected.

Reply:
I believe, this is the expected order, even with Logger-s. Where you specify Exception first, then formatted message.